### PR TITLE
Adds operationType to info parameter passed to resolver fn. Fixes #70.

### LINKED ISSUE #70: Operation type not provided
I cannot find a way to get the operation type when parsing GraphQL.

If I parse the following using graphql-tag:
```
gql`mutation {
  user (name:"John") {
    name
  }
}`
```
I get the following AST:
```
{ kind: 'Document',
  definitions:
   [ { kind: 'OperationDefinition',
       operation: 'mutation',
       name: undefined,
       variableDefinitions: [],
       directives: [],
       selectionSet: [Object] } ],
  loc: { start: 0, end: 55 } }
```

And if I log out all of the fields passed to my resolver, at no point does it tell me about the operation type.
```
export function resolver(fieldName, rootValue, args, context, info) {
  console.log(fieldName, rootValue, args, info)
  return {}
}
```

I get the following output:
```
users {} { name: 'John' } { isLeaf: false, resultKey: 'users' }
name {} null { isLeaf: true, resultKey: 'name' }
```

It would be really useful if the AST definition was passed into the Info, that way nothing gets lost.

### PR DESCRIPTION
**Pull request by calummoore**

### CODE REVIEW

File: src/graphql.ts [hunk 1]
```diff
@@ -46,12 +47,14 @@ export type ExecContext = {
   resultMapper: ResultMapper;
   resolver: Resolver;
   fragmentMatcher: FragmentMatcher;
+  operation: string;
```
Comment (stubailo) [inline src/graphql.ts]: Can we call it `operationType` everywhere?

File: src/graphql.ts [hunk 2]
```diff
@@ -77,6 +80,7 @@ export function graphql(
   execOptions: ExecOptions = {},
 ) {
   const mainDefinition = getMainDefinition(document);
+  const operation = (mainDefinition as OperationDefinitionNode).operation;
```
Comment (stubailo) [inline src/graphql.ts]: I think it would be nice if this worked for a fragment as well, setting the `operationType` to `fragment` in that case.

**Comment by jbaxleyiii** [comment]
I wanted to let everyone know that in the move to graphql-anywhere falling under the apollo-client repo, I will be taking a look at all of these PR's and may ask for you to recreate them in the new repo. I'm SO sorry this has been open for a while, one of the reasons for the move is to make sure PR's like this don't get missed by our team.

Thanks so much for contributing to this project btw! We want to continue to make it better and better with your help 🎉

**Comment by stubailo** [comment]
This repository has been moved into Apollo client, here: https://github.com/apollographql/apollo-client/tree/master/packages/graphql-anywhere

I'm going to shut down this repo, if you could reopen the PR there, that would be much appreciated!

**Pull request closed without merging**